lib/galaxy/web/base/interactive_environments.py

```diff
@@ -227,14 +238,23 @@ def docker_cmd(self, image, env_override={}, volumes=[]):
         # Then we format in the entire docker command in place of
         # {docker_args}, so as to let the admin not worry about which args are
         # getting passed
-        command = command.format(docker_args='run {command_inject} {environment} -d -P {import_volume_str} {volume_str} {image}')
+        command_inject = self.attr.viz_config.get("docker", "command_inject")
+        if not self.attr.swarm_mode:
+            command = command.format(docker_args='run {command_inject} {environment} -d -P {import_volume_str} {volume_str} {image}')
+            # --sig-proxy is not valid in the swarm mode context
+            if '--sig-proxy=' not in self.attr.command_inject:
```

Comment:
You are not comparing here `command_inject` is this correct? In line 252 you not passing in `command_inject`. So I'm not sure this is used.